packages/@webex/media-helpers/src/webrtc-core.ts

```diff
@@ -29,18 +31,22 @@ export type ServerMuteReason =
   | 'localUnmuteRequired'; // server forced the client to be unmuted
 
 // these events are in addition to WCME events. This will be properly typed once webrtc-core event types inheritance is fixed
-export enum LocalMicrophoneTrackEvents {
+export enum LocalMicrophoneStreamEventNames {
   ServerMuted = 'muted:byServer',
 }
 
 // these events are in addition to WCME events. This will be properly typed once webrtc-core event types inheritance is fixed
-export enum LocalCameraTrackEvents {
+export enum LocalCameraStreamEventNames {
   ServerMuted = 'muted:byServer',
 }
 
-export class LocalMicrophoneTrack extends WcmeLocalMicrophoneTrack {
+export class LocalMicrophoneStream extends WcmeLocalMicrophoneStream {
   private unmuteAllowed = true;
 
+  [LocalMicrophoneStreamEventNames.ServerMuted] = new TypedEvent<
```

Comment:
don't we need to also use `AddEvents` and export the type created with `WithEventsDummyType`? similarily like it's done in webrtc-core when classes emitting events extend other classes emitting events? see here: https://github.com/webex/webrtc-core/blob/787e429d070ac40aa8a817eeb2eee21a6fe1e72c/src/media/local-stream.ts#L193

Reply:
I am still not very familiar with Typed events. I'll leave it to Bryce to answer.

CC - @brycetham

Reply:
Oh, that's a good question. Yeah it seems like we should, based on https://github.com/webex/ts-events#methodology.

Perhaps @bbaldino can explain better why this is necessary.

Reply:
Yeah if you want the 'js-style' syntax (`class.on('eventName', handler)`) you'll want to declare the events as laid out in the ts-events README.  Using a raw `TypedEvent` gets you something like `class.event.on(handler)`

Reply:
So, do we need to change this to a Typed event?

@brycetham @marcin-bazyl

Reply:
we don't want to use just a raw TypedEvent, we want to be able to do class.on(event, handler), see README from ts-events lib and webrtc-core source code for an example how to do it, look at `_LocalStream` class - it extends Stream class and add more events to it, we are doing the same here, we're extending webrtc-core classes and add events to them.

Reply:
Done in #3053